File: frontend/src/types/focus.ts
// types/focus.ts

/** 집중 방식. stopwatch는 계속 누적, pomodoro는 25분/5분을 반복 */
export type FocusMode = 'stopwatch' | 'pomodoro';

/** 뽀모도로 단계 */
export type PomodoroPhase = 'focus' | 'break';

/** 오늘의 집중 요약 */
export interface FocusSummary {
  /** 오늘 누적 집중 시간(초) */
  focusedSeconds: number;
  /** 오늘 목표 시간(분). 텐미닛 플래너의 계획 시간 합계로 정한다 */
  targetMinutes: number;
  /** 연속 달성일. 투두가 체크된 날을 세며 서버가 계산한다 */
  streakDays: number;
  /** 지금까지의 최고 연속 기록(일) */
  bestStreakDays: number;
  /** 지금 연속 기록이 최고 기록인지 */
  isBestStreak: boolean;
}

/** 서버에 남기는 집중 세션 한 건 */
export interface FocusSessionInput {
  mode: FocusMode;
  /** 집중한 시간(초) */
  seconds: number;
  /** 시작 시각 (ISO 8601) */
  startedAt: string;
}

/** 스톱워치 한 바퀴 = 60분 */
export const STOPWATCH_ROUND_MINUTES = 60;
/** 뽀모도로 집중 시간(분) */
export const POMODORO_FOCUS_MINUTES = 25;
/** 뽀모도로 휴식 시간(분) */
export const POMODORO_BREAK_MINUTES = 5;
/** 긴 휴식 시간(분). 집중 4번마다 한 번 */
export const POMODORO_LONG_BREAK_MINUTES = 15;
/** 몇 번 집중할 때마다 긴 휴식을 줄지 */
export const POMODORO_LONG_BREAK_EVERY = 4;

File: backend/app/schemas/focus.py
"""집중 스키마 (api.md §6).

프론트가 실제 쓰는 필드는 FocusSummary{focusedSeconds,targetMinutes},
WeeklyFocus{hours,diffFromLastWeek}. streak 3종은 api.md 스펙(프론트는 무시해도 안전).
"""

from datetime import datetime

from pydantic import Field

from app.schemas.common import CamelModel


class FocusSummaryOut(CamelModel):
    focused_seconds: int
    target_minutes: int
    # 연속 기록(투두 체크된 날 기준). 과거 전체 이력으로 서버가 계산.
    streak_days: int = 0
    best_streak_days: int = 0
    is_best_streak: bool = False


class FocusSessionIn(CamelModel):
    mode: str  # stopwatch | pomodoro
    seconds: int = Field(gt=0)
    started_at: datetime
    goal_id: str | None = None  # 목표별 집계용(선택)


class WeeklyFocusOut(CamelModel):
    hours: list[float]  # 월~일 7개(시간 단위)
    diff_from_last_week: float


class DailyFocusItem(CamelModel):
    date: str  # YYYY-MM-DD
    seconds: int


class DailyFocusOut(CamelModel):
    days: list[DailyFocusItem]
